qiita_db/commands.py

```diff
@@ -300,3 +304,101 @@ def load_parameters_from_cmd(name, fp, table):
                          "The format is PARAMETER_NAME<tab>VALUE")
 
     return constructor.create(name, **params)
+
+
+def update_preprocessed_data_from_cmd(sl_out_dir, study_id):
+    """Updates the preprocessed data of the study 'study_id'
+
+    Parameters
+    ----------
+    sl_out_dir : str
+        The path to the split libraries output directory
+    study_id : int
+        The study_id of the study to be updated
+
+    Returns
+    -------
+    qiita_db.PreprocessedData
+        The updated preprocessed data
+
+    Raises
+    ------
+    ValueError
+        If sl_out_dir does not contain all the required files
+        If the study does not have any preprocessed data
+
+    Notes
+    -----
+    If the study has more than one preprocessed data, the one with lower id
```

Comment:
I see your point. This is quite useful for the current use case (updating the EMP studies where we are sure that the preprocessed data that we want to modify is the one with the lower id). However, for a more general use case, this can be dangerous. What about adding an optional parameter `preprocessed_data_id` that, if passed, it updates that one instead that the one with the lowest id?